src/setup-java.ts

```diff
@@ -23,8 +23,15 @@ async function run() {
       core.getInput('server-username', {required: false}) || undefined;
     const password =
       core.getInput('server-password', {required: false}) || undefined;
+    const overwriteSettings =
```

Comment:
I suggest to use the same approach like `actions/checkout`: https://github.com/actions/checkout/blob/main/src/input-helper.ts#L118
May be we can wrap it in small util func